tests/components/mqtt_eventstream/test_init.py

```diff
@@ -25,177 +23,170 @@ def mock_storage(hass_storage):
-
-    @patch("homeassistant.components.mqtt.async_publish")
-    def test_wrong_ignored_event_sends_over_stream(self, mock_pub):
-        """Test the ignoring of sending events if defined."""
-        assert self.add_eventstream(pub_topic="bar", ignore_event=["statee_changed"])
-        self.hass.block_till_done()
-
-        e_id = "entity.test_id"
-        event = {}
-        event["event_type"] = EVENT_STATE_CHANGED
-        new_state = {"state": "on", "entity_id": e_id, "attributes": {}}
-        event["event_data"] = {"new_state": new_state, "entity_id": e_id}
+        mock_state_change_event(hass, State(e_id, "on"))
+        await hass.async_block_till_done()
+        await hass.async_block_till_done()
+
+    # The order of the JSON is indeterminate,
+    # so first just check that publish was called
+    mqtt_mock.async_publish.assert_called_with(pub_topic, ANY, 0, False)
+    assert mqtt_mock.async_publish.called
+
+    # Get the actual call to publish and make sure it was the one
+    # we were looking for
+    msg = mqtt_mock.async_publish.call_args[0][1]
+    event = {}
+    event["event_type"] = EVENT_STATE_CHANGED
+    new_state = {
+        "last_updated": now.isoformat(),
+        "state": "on",
+        "entity_id": e_id,
+        "attributes": {},
+        "last_changed": now.isoformat(),
+    }
+    event["event_data"] = {"new_state": new_state, "entity_id": e_id}
+
+    # Verify that the message received was that expected
+    result = json.loads(msg)
+    result["event_data"]["new_state"].pop("context")
+    assert result == event
+
+
+async def test_time_event_does_not_send_message(hass, mqtt_mock):
+    """Test the sending of a new message if time event."""
+    assert await add_eventstream(hass, pub_topic="bar")
+    await hass.async_block_till_done()
+
+    # Reset the mock because it will have already gotten calls for the
+    # mqtt_eventstream state change on initialization, etc.
+    mqtt_mock.async_publish.reset_mock()
+
+    async_fire_time_changed(hass, dt_util.utcnow())
+    assert not mqtt_mock.async_publish.called
+
+
+async def test_receiving_remote_event_fires_hass_event(hass, mqtt_mock):
+    """Test the receiving of the remotely fired event."""
+    sub_topic = "foo"
+    assert await add_eventstream(hass, sub_topic=sub_topic)
+    await hass.async_block_till_done()
+
+    calls = []
+
+    @callback
+    def listener(_):
+        calls.append(1)
+
+    hass.bus.async_listen_once("test_event", listener)
+    await hass.async_block_till_done()
+
+    payload = json.dumps(
+        {"event_type": "test_event", "event_data": {}}, cls=JSONEncoder
+    )
+    async_fire_mqtt_message(hass, sub_topic, payload)
+    await hass.async_block_till_done()
+
+    assert 1 == len(calls)
+
+
+async def test_ignored_event_doesnt_send_over_stream(hass, mqtt_mock):
+    """Test the ignoring of sending events if defined."""
+    assert await add_eventstream(hass, pub_topic="bar", ignore_event=["state_changed"])
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5745133)